### PR TITLE
Use local docker file for builds

### PR DESCRIPTION
**Pull request by Que3216**
Uses the local Dockerfile rather than the published one when running tests. This ensures that the tests are actually testing the code on the PR, rather than the last published code.

### CODE REVIEW

File: scripts/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (leafty) [inline scripts/package.json:1]: I think I would do it without `package.json` and `yarn.lock`.

**Review by leafty — APPROVED** [review]

File: scripts/package.json (hunk 1 above)
Comment (Que3216) [reply scripts/package.json:1]: And just a `yarn install yaml` command?

Comment (leafty) [reply scripts/package.json:1]: I would be OK installing all the dependencies, we can have the workflow share caches for `node_modules`.

Comment (Que3216) [reply scripts/package.json:1]: But doesn't the main `yarn install` happen inside docker when building the docker image? How easy would it be to share caches for that?

Comment (Que3216) [reply scripts/package.json:1]: I guess we could do the yarn install outside the image, and then COPY the node_modules in

Comment (leafty) [reply scripts/package.json:1]: It would not work when running with `image: "Dockerfile"` with the `COPY` of `node_modules`.

Comment (Que3216) [reply scripts/package.json]: I've updated -- it seems to work (I'm yarn install'ing in both publish task & run-all-tests task)

**Review by leafty — APPROVED** [review]
LGTM great! :clap:

**Pull request merged**